### PR TITLE
feat: contextual primary colors & primary pill patterns

### PR DESCRIPTION
**Pull request by pretzelhammer**
## Describe the problem this PR addresses

no contextual primary colors or primary pill patterns

## Describe the changes in this PR

- adds generation of `contextualPrimary` colors to `makerColors` util
- adds new `primary`, `primaryOutline`, and `primarySubtle` pill patterns to default theme
- exports all contextual colors as CSS vars from MTheme
- updates MTheme docs to explain/describe all newly exported color CSS vars
- creates new `Discounts` lab experiment which shows how to make discount-y looking pills, notices, & cards
- updates `ThemeTest` lab to use updated `makerColors` util to generate the contextual primary colors

## Other information

Design has reviewed the updates and has given them the thumbs-up

- [Discounts lab](https://square.github.io/maker/lab/discounts/#/Discounts)
- [ThemeTest lab](https://square.github.io/maker/lab/discounts/#/ThemeTest)
- [MPill docs](https://square.github.io/maker/styleguide/discounts/#/Pill)
- [MTheme docs](https://square.github.io/maker/styleguide/discounts/#/Theme)

### CODE REVIEW

**Comment by landondurnan** [comment]
We likely need some design insight for dark primary colors. When using a dark primary like black, we may want to flip the contrast to ensure elements like icons stand out
CleanShot 2022-06-17 at 09 48 39
.

File: src/components/Theme/README.md [hunk 1]
```diff
@@ -723,20 +724,48 @@ The Theme component makes these reusable CSS variables available to all DOM node
...
---maker-color-elevation | elevation color
---maker-color-overlay | overlay color
+--maker-color-elevation | elevation color, e.g. color of elevated elements, like ActionBarButtons or selected Segments in SegmentedControl
+--maker-color-overlay | overlay color, e.g. color of Modal layer, Dialog layer, etc
+
+#### Contextual colors
+
+Variable | Purpose
+-|-
+--maker-color-primary-fill | primary fill color, same as primary color
```
Comment (laurenhu) [inline src/components/Theme/README.md:753]: I'm not sure if i want to make contextual colors publicly available yet. The logic for how they're applied to Maker components and how they react over background is not super intuitive and I would prefer to just make add success or critical variants for a component that doesn't currently support those patterns than have engineers try to make sense of our generated contextual colors.

File: src/components/Theme/src/Theme.vue [hunk 2]
```diff
@@ -123,24 +123,49 @@ export default {
...
 				'--maker-color-neutral-100': colors['neutral-100'],
+
+				// general colors
 				'--maker-color-primary': colors.primary,
 				'--maker-color-background': colors.background,
 				'--maker-color-heading': colors.heading,
 				'--maker-color-body': colors.body,
 				'--maker-color-elevation': colors.elevation,
 				'--maker-color-overlay': colors.overlay,
+
```
Comment (laurenhu) [inline src/components/Theme/src/Theme.vue:141]: See above - I'm leery of adding contextual colors to the public Maker color variables.

**Review by laurenhu — COMMENTED** [review]
Everything looks good and the primary contextual colors look/work great and feel consistent with our other. My only feedback is about exposing the contextual colors as Maker color variables.

**Review by laurenhu — APPROVED** [review]

**Pull request merged**